src/io/windows.zig

```diff
@@ -0,0 +1,1132 @@
+const std = @import("std");
+const os = std.os;
+const assert = std.debug.assert;
+const log = std.log.scoped(.io);
+const config = @import("../config.zig");
+
+const FIFO = @import("../fifo.zig").FIFO;
+const Time = @import("../time.zig").Time;
+const buffer_limit = @import("../io.zig").buffer_limit;
+
+pub const IO = struct {
+    iocp: os.windows.HANDLE,
+    timer: Time = .{},
+    io_pending: usize = 0,
+    timeouts: FIFO(Completion) = .{},
+    completed: FIFO(Completion) = .{},
+
+    pub fn init(entries: u12, flags: u32) !IO {
+        _ = entries;
+        _ = flags;
+
+        _ = try os.windows.WSAStartup(2, 2);
+        errdefer os.windows.WSACleanup() catch unreachable;
+
+        const iocp = try os.windows.CreateIoCompletionPort(os.windows.INVALID_HANDLE_VALUE, null, 0, 0);
+        return IO{ .iocp = iocp };
+    }
+
+    pub fn deinit(self: *IO) void {
+        assert(self.iocp != os.windows.INVALID_HANDLE_VALUE);
+        os.windows.CloseHandle(self.iocp);
+        self.iocp = os.windows.INVALID_HANDLE_VALUE;
+
+        os.windows.WSACleanup() catch unreachable;
+    }
+
+    pub fn tick(self: *IO) !void {
+        return self.flush(.non_blocking);
+    }
+
+    pub fn run_for_ns(self: *IO, nanoseconds: u63) !void {
+        const Callback = struct {
+            fn onTimeout(timed_out: *bool, completion: *Completion, result: TimeoutError!void) void {
+                _ = result catch unreachable;
+                _ = completion;
+                timed_out.* = true;
+            }
+        };
+
+        var timed_out = false;
+        var completion: Completion = undefined;
+        self.timeout(*bool, &timed_out, Callback.onTimeout, &completion, nanoseconds);
+
+        while (!timed_out) {
+            try self.flush(.blocking);
+        }
+    }
+
+    const FlushMode = enum {
+        blocking,
+        non_blocking,
+    };
+
+    fn flush(self: *IO, mode: FlushMode) !void {
+        if (self.completed.peek() == null) {
+            // Compute how long to poll by flushing timeout completions.
+            // NOTE: this may push to completed queue
+            var timeout_ms: ?os.windows.DWORD = null;
+            if (self.flush_timeouts()) |expires_ns| {
+                // 0ns expires should have been completed not returned
+                assert(expires_ns != 0); 
```

Comment:
Great to see this assertion here. Are there any other places across Linux/Darwin/Windows where we could add assertions in the control plane?